tests/crypto_test.py

```diff
@@ -26,3 +56,286 @@ def test_returned_string_is_lower_case(self):
+        self.assertEqual(expected, actual)
+
+    def test_invalid_key(self):
+        with self.assertRaises(GPGProblem) as caught:
+            crypto.get_key('z')
+        self.assertEqual(caught.exception.code, GPGCode.NOT_FOUND)
+
+    @mock.patch('alot.crypto.check_uid_validity', mock.Mock(return_value=True))
+    def test_signed_only_true(self):
+        try:
+            crypto.get_key(self.FPR, signed_only=True)
+        except GPGProblem as e:
+            raise AssertionError(e)
+
+    @mock.patch('alot.crypto.check_uid_validity', mock.Mock(return_value=False))
+    def test_signed_only_false(self):
+        with self.assertRaises(GPGProblem) as e:
+            crypto.get_key(self.FPR, signed_only=True)
+        self.assertEqual(e.exception.code, GPGCode.NOT_FOUND)
+
+    @staticmethod
+    def _context_mock():
+        error = gpgme.GpgmeError()
+        error.code = gpgme.ERR_AMBIGUOUS_NAME
+        context_mock = mock.Mock()
+        context_mock.get_key = mock.Mock(side_effect=error)
+
+        return context_mock
+
+    def test_ambiguous_one_valid(self):
+        invalid_key = make_key(invalid=True)
+        valid_key = make_key()
+
+        with mock.patch('alot.crypto.gpgme.Context',
+                        mock.Mock(return_value=self._context_mock())), \
+                mock.patch('alot.crypto.list_keys',
+                           mock.Mock(return_value=[valid_key, invalid_key])):
+            key = crypto.get_key('placeholder')
+        self.assertIs(key, valid_key)
+
+    def test_ambiguous_two_valid(self):
+        with mock.patch('alot.crypto.gpgme.Context',
+                        mock.Mock(return_value=self._context_mock())), \
+                mock.patch('alot.crypto.list_keys',
+                           mock.Mock(return_value=[make_key() for _ in xrange(2)])):
+            with self.assertRaises(crypto.GPGProblem) as cm:
+                crypto.get_key('placeholder')
+        self.assertEqual(cm.exception.code, GPGCode.AMBIGUOUS_NAME)
+
+    def test_ambiguous_no_valid(self):
+        with mock.patch('alot.crypto.gpgme.Context',
+                        mock.Mock(return_value=self._context_mock())), \
+                mock.patch('alot.crypto.list_keys',
+                           mock.Mock(return_value=[make_key(invalid=True)
+                                                   for _ in xrange(2)])):
+            with self.assertRaises(crypto.GPGProblem) as cm:
+                crypto.get_key('placeholder')
+        self.assertEqual(cm.exception.code, GPGCode.NOT_FOUND)
+
+
+class TestEncrypt(_TestKey):
+
+    def test_encrypt(self):
+        to_encrypt = "this is a string\nof data."
+        encrypted = crypto.encrypt(to_encrypt, keys=[crypto.get_key(self.FPR)])
+
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(encrypted)
+            enc_file = f.name
+        self.addCleanup(os.unlink, enc_file)
+
+        dec = subprocess.check_output(['gpg', '--decrypt', enc_file])
```

Comment:
Again is there a reason you don't use [`gpgme.Context.decrypt`](http://pygpgme.readthedocs.io/en/latest/api.html#gpgme.Context.decrypt)?